crates/test-programs/wasi-tests/src/lib.rs

```diff
@@ -43,3 +44,25 @@ pub unsafe fn create_file(dir_fd: wasi::Fd, filename: &str) {
     );
     wasi::fd_close(file_fd).expect("closing a file");
 }
+
+// Returns: (rights_base, rights_inheriting)
+pub unsafe fn fd_get_rights(fd: wasi::Fd) -> (wasi::Rights, wasi::Rights) {
+    let fdstat = wasi::fd_fdstat_get(fd).expect("fd_fdstat_get failed");
+    (fdstat.fs_rights_base, fdstat.fs_rights_inheriting)
+}
+
+pub unsafe fn drop_rights(
+    fd: wasi::Fd,
+    drop_base: wasi::Rights,
+    drop_inheriting: wasi::Rights,
+) {
+    let (current_base, current_inheriting) = fd_get_rights(fd);
+
+    let new_base = current_base & !drop_base;
+    let new_inheriting = current_inheriting & !drop_inheriting;
+
+    assert!(
+        wasi::fd_fdstat_set_rights(fd, new_base, new_inheriting).is_ok(),
+        "dropping fd rights",
+    );
+}
```

Comment:
You could probably make it a little clearer by using an `expect` rather than `assert` on `Ok` :-)

Reply:
Yes, but then there will be no line info displayed in case of a panic. (I know that I'm using `expect` in other places but it's worth discussing)

Reply:
Done.

Reply:
Apropos of this: https://github.com/rust-lang/rust/pull/67887 just landed, so this issue should go away once that makes its way to stable rust :-).